feat: add an ability to let users sign up

Amirbapiri opened the pull request:
This feature provides an ability to let users sign up.

Amirbapiri committed:
feat: add an ability to let users sign up

This feature provides an ability to let users sign up.

mehdy commented on app.py:
```diff
@@ -161,3 +163,12 @@ def mail_most_voted(title):
+def register_user():
+    data = request.json
+    if data not in users:
+        users.append(data)
```
The problem here is that you're accepting any data you'd receive.

Amirbapiri committed:
fix: validate registration data

The data needs to be validated before insert it to the DB or for other
usages.
Here we assume 'id' and 'email' are unique for each user.
